src/scheduler.cpp

```diff
@@ -114,6 +114,27 @@ void CScheduler::scheduleFromNow(CScheduler::Function f, int64_t deltaMilliSecon
     schedule(f, boost::chrono::system_clock::now() + boost::chrono::milliseconds(deltaMilliSeconds));
 }
 
+void CScheduler::MockForward(boost::chrono::seconds delta_seconds)
+{
+    {
+        boost::unique_lock<boost::mutex> lock(newTaskMutex);
+
+        // use temp_queue to maintain updated schedule
+        std::multimap<boost::chrono::system_clock::time_point, Function> temp_queue;
+
+        for (const auto& element : taskQueue){
+            boost::chrono::system_clock::time_point new_time = element.first - delta_seconds;
```

Comment:
May want to make sure that element.first > delta_seconds

Reply:
or maybe not? What does negative time do here...

Reply:
oh interesting. the first element is a time point, but I'm not sure what would happen if you pass in delta_seconds big enough to actually make it negative. I'll check it out.